src/components/Header/HeaderTabletAndDesktop/HeaderNavigation.tsx

```diff
@@ -1,116 +1,111 @@
-                    : "",
-                )}
-                href={href}
-                onClick={(e) => {
-                  if (subMenuData) {
-                    e.preventDefault();
-                    handleToggle(actualIndex);
-                  } else {
-                    setOpenIndex(null);
-                  }
-                }}
-                aria-label={`Przejdź do ${label}`}
-                aria-expanded={
-                  subMenuData ? openIndex === actualIndex : undefined
-                }
-                aria-haspopup={subMenuData ? "menu" : undefined}
-              >
-                {label}
-                {subMenuData && (
-                  <IconChevronDown
-                    className={cn(
-                      "h-4 w-4 transition-transform duration-300 ease-in-out desktop:h-5 desktop:w-5",
-                      openIndex === actualIndex ? "rotate-180" : "rotate-0",
-                    )}
-                    aria-hidden="true"
-                  />
-                )}
-              </ActiveLink>
-              {subMenuData && openIndex === actualIndex && (
-                <ul
-                  className="absolute z-50 mt-2 flex w-[243px] flex-col gap-5 bg-greenB p-4 shadow-lg contrast:bg-yellowContrast"
-                  aria-label={`Podmenu dla ${label}`}
-                >
-                  {subMenuData.map(
-                    ({ label: subLabel, href: subHref }, subIndex) => (
-                      <li
-                        key={subIndex}
-                        className="mb-2"
-                        aria-label={`Podmenu: ${subLabel}`}
-                      >
-                        <ActiveLink
-                          href={subHref}
-                          className="text-16 text-white transition duration-300 ease-in-out hover:text-yellowVintage active:text-blue contrast:text-black00"
-                          onClick={handleSubMenuLinkClick}
-                          aria-label={`Przejdź do ${subLabel}`}
-                        >
-                          {subLabel}
-                        </ActiveLink>
-                      </li>
-                    ),
-                  )}
-                </ul>
-              )}
-            </li>
+              item={item}
+              index={index}
+              baseIndex={baseIndex}
+              isSubmenuOpen={isSubmenuOpen(actualIndex)}
+              toggleSubmenu={() => toggleSubmenu(actualIndex)}
+              closeSubmenu={closeSubmenu}
+            />
           );
         })}
       </ul>
     </nav>
   );
 };
+
+interface NavItemProps {
+  item: NavItem;
+  index: number;
+  baseIndex: number;
+  isSubmenuOpen: boolean;
+  toggleSubmenu: () => void;
+  closeSubmenu: () => void;
+}
+
+const NavItem = ({
+  item: { label, href },
+  index,
+  baseIndex,
+  isSubmenuOpen,
+  toggleSubmenu,
+  closeSubmenu,
+}: NavItemProps) => {
+  const actualIndex = baseIndex + index;
```

Comment:
This looks a little bit off. But it is something caused by the data we are using for navigation.
We can improve this in the future.